packages/p2p/src/components/my-ads/form-ads.jsx

```diff
@@ -5,60 +5,66 @@ import {
     Autocomplete,
     Dropdown,
     Loading,
+    Icon,
     Input,
     Button,
     ThemedScrollbars }                from '@deriv/components';
+import ObjectUtils                    from '@deriv/shared/utils/object';
+import Dp2pContext                    from 'Components/context/dp2p-context';
 import FooterActions                  from 'Components/footer-actions/footer-actions.jsx';
 import { localize }                   from 'Components/i18next';
 import PageReturn                     from 'Components/page-return/page-return.jsx';
+import { requestWS }                  from 'Utils/websocket';
 
 class FormAds extends Component {
     state = {
-        initial_values: {
-            country        : '',
-            currency       : '',
-            type           : '',
-            asset          : '',
-            fix_price      : '',
-            amount         : '',
-            min_transaction: '',
-            max_transaction: '',
-            advertiser_note: '',
-        },
-        is_loading: true,
+        country      : '',
+        error_message: '',
+        is_loading   : true,
     };
 
-    componentDidMount() {
-        // TODO: [p2p-fix-api] call get offer detail api and populate state
-        const new_initial_values = {
-            country        : 'Indonesia',
-            currency       : 'IDR',
-            type           : 'buy',
-            asset          : 'USD',
-            fix_price      : 10000,
-            amount         : 50,
-            min_transaction: 1000,
-        };
-        this.setState({
-            initial_values: new_initial_values,
-        });
+    async componentDidMount() {
+        const residence_response = await requestWS({ residence_list: 1 });
+        const current_residence = residence_response.residence_list.find(
+            (residence) => residence.value === this.context.residence
+        );
+        const display_residence = ObjectUtils.getPropertyValue(current_residence, 'text') || '';
 
+        // TODO: [p2p-fix-api] call get offer detail api and populate state
+        // this is pending offer edit enabled in the API
         if (this.props.ad_id) {
             // call the api, get the file based on id
             // populate the state from the respnose
+            this.setState({
+                country: display_residence,
+            });
         } else {
-            this.setState({ is_loading: false });
+            this.setState({ is_loading: false, country: display_residence });
         }
     }
 
-    handleSubmit(formik_vars, { setSubmitting }) {
-        // TODO: [p2p-fix-api] call offer create api
-        // eslint-disable-next-line no-console
-        console.log(this.state);
-        // eslint-disable-next-line no-console
-        console.log(formik_vars);
-        setSubmitting(false);
-    }
+    handleSubmit = (values, { setSubmitting }) => {
+        this.setState({ error_message: '' });
```

Comment:
think we need to add `setSubmitting(true);` here

Reply:
@oskar-binary it's automatically set to true on click of submit of the form